substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -2229,3 +2196,24 @@ impl<T: Config> ChangeNode<LocationOf<T>, PubConfigOf<T>, InterfaceOf<T>, Serial
         }
     }
 }
+
+impl<T: Config> PublicIpModifier for Pallet<T> {
+    fn ip_removed(ip: &PublicIP) {
+        if let Some(mut deployment) = Deployments::<T>::get(ip.contract_id) {
+            // Todo remove public ip from deployment and capacity contract
```

Comment:
This can be removed right? As we are going to remove it from the farm objects and keep it in the deployments instead.